src/components/course/data/courseLoader.js

```diff
@@ -64,7 +62,21 @@ export default function makeCourseLoader(queryClient) {
       queryClient.ensureQueryData(queryCouponCodeRequests(enterpriseCustomer.uuid, authenticatedUser.email)),
       queryClient.ensureQueryData(queryBrowseAndRequestConfiguration(enterpriseCustomer.uuid)),
     ]);
-
+    const redeemableLearnerCreditPoliciesLoader = await queryClient.ensureQueryData(queryRedeemablePolicies({
+      enterpriseUuid: enterpriseCustomer.uuid,
+      lmsUserId: authenticatedUser.userId,
+    }));
+    const {
+      allocatedCourseRunAssignmentKeys,
+      hasAssignedCourseRuns,
+      hasMultipleAssignedCourseRuns,
+    } = determineAllocatedCourseRuns({
+      courseKey,
+      redeemableLearnerCreditPolicies: redeemableLearnerCreditPoliciesLoader,
+    });
+    if (!courseRunKey && hasAssignedCourseRuns) {
+      courseRunKey = hasMultipleAssignedCourseRuns ? null : allocatedCourseRunAssignmentKeys[0];
```

Comment:
[nit] might be worth adding a helper comment to explain why we return `null` for `hasMultipleAssignedCourseRuns`, something to the effect of:

```jsx
// only override `courseRunKey` when learner has a single allocated assignment
```